src/Database/Postgres.purs

```diff
@@ -1,153 +1,186 @@
 module Database.Postgres
   ( Query(..)
-  , Client()
-  , DB()
-  , ConnectionInfo()
-  , ConnectionString()
-  , mkConnectionString
+  , Client
+  , Pool
+  , DB
+  , ConnectionInfo
+  , ClientConfig
+  , PoolConfig
+  , ConnectionString
+  , connectionInfoFromConfig
+  , connectionInfoFromString
+  , defaultPoolConfig
+  , mkDefaultPoolConfig
   , connect
-  , disconnect
+  , release
   , end
   , execute, execute_
   , query, query_
   , queryValue, queryValue_
   , queryOne, queryOne_
-  , withConnection
-  , withClient
+  , withPool
+  , mkPool
   ) where
 
 import Prelude
+
+import Control.Monad.Aff (Aff, bracket)
+import Control.Monad.Aff.Compat (EffFnAff, fromEffFnAff)
 import Control.Monad.Eff (kind Effect, Eff)
-import Data.Either (Either, either)
-import Data.Function.Uncurried (Fn2(), runFn2)
+import Control.Monad.Eff.Class (liftEff)
+import Control.Monad.Eff.Exception (error)
+import Control.Monad.Error.Class (throwError)
+import Control.Monad.Except (runExcept)
 import Data.Array ((!!))
+import Data.Either (Either, either)
 import Data.Foreign (Foreign, MultipleErrors)
 import Data.Foreign.Class (class Decode, decode)
 import Data.Maybe (Maybe(Just, Nothing), maybe)
-import Control.Monad.Except (runExcept)
-import Control.Monad.Aff (Aff, finally)
-import Control.Monad.Eff.Class (liftEff)
-import Control.Monad.Eff.Exception (error)
-import Control.Monad.Error.Class (throwError)
 import Data.Traversable (sequence)
-
 import Database.Postgres.SqlValue (SqlValue)
+import Unsafe.Coerce (unsafeCoerce)
 
 newtype Query a = Query String
 
+foreign import data Pool :: Type
+
 foreign import data Client :: Type
 
 foreign import data DB :: Effect
 
+foreign import data ConnectionInfo :: Type
+
 type ConnectionString = String
 
-type ConnectionInfo =
+connectionInfoFromString :: ConnectionString -> ConnectionInfo
+connectionInfoFromString s = unsafeCoerce { connectionString: s }
+
+type ClientConfig =
   { host :: String
-  , db :: String
+  , database :: String
   , port :: Int
   , user :: String
   , password :: String
+  , ssl :: Boolean
+  }
+
+type PoolConfig =
+  { connectionTimeoutMillis :: Int
+  , idleTimeoutMillis :: Int
+  , max :: Int
+  }
+
+defaultPoolConfig :: PoolConfig
+defaultPoolConfig =
+  { connectionTimeoutMillis: 0
+  , idleTimeoutMillis: 30000
+  , max: 10
+  }
+
+connectionInfoFromConfig :: ClientConfig -> PoolConfig -> ConnectionInfo
+connectionInfoFromConfig c p = unsafeCoerce
+  { host: c.host
+  , database: c.database
+  , port: c.port
+  , user: c.user
+  , password: c.password
+  , ssl: c.ssl
+  , connectionTimeoutMillis: p.connectionTimeoutMillis
+  , idleTimeoutMillis: p.idleTimeoutMillis
+  , max: p.max
   }
 
-mkConnectionString :: ConnectionInfo -> ConnectionString
-mkConnectionString ci =
-    "postgres://"
-  <> ci.user <> ":"
-  <> ci.password <> "@"
-  <> ci.host <> ":"
-  <> show ci.port <> "/"
-  <> ci.db
+mkDefaultPoolConfig :: ClientConfig -> ConnectionInfo
```

Comment:
Let's actually remove this for now. I think the API of calling either `connectionInfoFromString ...` or `connectionInfoFromConfig {...} defaultPoolConfig` is good enough.